source/containers.tex

```diff
@@ -25486,82 +25634,76 @@
 \begin{itemdecl}
 template<class IndexType, size_t... Extents, class... SliceSpecifiers>
   constexpr auto submdspan_extents(const extents<IndexType, Extents...>& src,
-                                   SliceSpecifiers... slices);
+                                   SliceSpecifiers... raw_slices);
 \end{itemdecl}
 
 \begin{itemdescr}
+\pnum
+Let \tcode{slices} be the pack intoduced by the following declaration:
+\begin{codeblock}
+auto [...slices] = submdspan_canonicalize_slices(src, raw_slices...);
+\end{codeblock}
+
 \pnum
 \constraints
-\tcode{sizeof...(slices)} equals \tcode{sizeof...(Extents)}.
+%FIXME: The paper cites "equals Extents::rank()" as the pre-existing wording.
+%Investigate whether the change is still okay (which only affects the left side).
+\tcode{sizeof...(SliceSpecifiers)} equals \tcode{sizeof...(Extents)}.
 
 \pnum
 \mandates
-For each rank index $k$ of \tcode{src.extents()},
-exactly one of the following is true:
+For each rank index $k$ of \tcode{src}:
 \begin{itemize}
-\item $S_k$ models \tcode{\libconcept{convertible_to}<IndexType>},
-\item $S_k$ models \tcode{\exposconcept{index-pair-like}<IndexType>},
-\item \tcode{is_convertible_v<$S_k$, full_extent_t>} is \tcode{true}, or
-\item $S_k$ is a specialization of \tcode{strided_slice}.
+\item
+  \tcode{SliceSpecifiers...[k]}
+  is a \tcode{submdspan} slice type for \tcode{IndexType}, and
+\item
+  \tcode{decltype(slices...[k])} is a valid \tcode{submdspan} slice type
+  for the $k^\text{th}$ extent of \tcode{extents<IndexType, Extents...>}.
 \end{itemize}
 
 \pnum
 \expects
-For each rank index $k$ of \tcode{src.extents()},
-all of the following are \tcode{true}:
-\begin{itemize}
-\item
-if $S_k$ is a specialization of \tcode{strided_slice}
-  \begin{itemize}
-  \item $\tcode{$s_k$.extent} = 0$, or
-  \item $\tcode{$s_k$.stride} > 0$
-  \end{itemize}
-\item
-$0 \le \tcode{\exposid{first_}<IndexType, $k$>(slices...)}$
-$\le \tcode{\exposid{last_}<$k$>(src, slices...)}$
-$\le \tcode{src.extent($k$)}$
-\end{itemize}
+For each rank index $k$ of \tcode{src},
+\tcode{slices...[k]} is a valid \tcode{submdspan} slice
+for the $k^\text{th}$ extent of \tcode{src}.
 
 \pnum
 Let \tcode{SubExtents} be a specialization of \tcode{extents} such that:
 
 \begin{itemize}
 \item
-%FIXME: I think we want the count here, "number" is ambiguous.
-\tcode{SubExtents::rank()} equals the number of $k$ such that
-$S_k$ does not model \tcode{\libconcept{convertible_to}<IndexType>}; and
+\tcode{SubExtents::rank()} equals
+\tcode{\exposid{MAP_RANK}(slices, Extents::rank())}; and
```

Comment:
I think we have the same issue as above here with `Extents::rank()`.  The existing text below says "`for each rank index $k$ of \tcode{Extents}`", so it may pay to introduce a name for the extents type.  For example, we could change paragraph 1 like this

> Let `E` denote `extents<IndexType, Extents...>`, and let `slices` be the pack ....

and then use `E` throughout, e.g., in paragraph 2.

That being said, perhaps it's better just to merge the text as-is and then file an LWG issue with the above as a proposed fix.  What do you think?